core/src/main/java/com/bumble/appyx/core/node/ParentNode.kt

```diff
@@ -212,6 +218,37 @@ abstract class ParentNode<Routing : Any>(
         }
     }
 
+    protected suspend inline fun <reified T : Node> attachWorkflow(
+        crossinline action: () -> Unit
+    ): T = withContext(lifecycleScope.coroutineContext) {
+        action()
+
+        // after executing action waiting for the children to sync with routing source and
+        // throw an exception after short timeout if desired child was not found
+        val result = withTimeoutOrNull(AttachWorkFlowSyncTimeout) {
+            waitForChildAttached<T>()
+        }
+        result ?: throw IllegalStateException(
+            "Expected child of type [${T::class.java}] was not found after executing action. " +
+                "Check that your action actually results in the expected child. "
+        )
+    }
+
+    protected suspend inline fun <reified T : Node> waitForChildAttached(): T =
+        suspendCancellableCoroutine { continuation ->
+            lifecycleScope.launch {
+                children.collect { childMap ->
+                    val childNodes = childMap.entries.mapNotNull { it.value.nodeOrNull }
+                    val childNodesOfExpectedType = childNodes.filterIsInstance<T>()
```

Comment:
It can be implemented without creation of 2 new lists:
```kotlin
childMap.entries.findLast { it is T } as T?
```

Also `last()` does not really makes sense because `Map.entries` is a `Set`.